winpty/src/windows.rs

```diff
@@ -283,25 +286,28 @@ impl SpawnConfig {
         spawnflags: SpawnFlags,
         appname: Option<&str>,
         cmdline: Option<&str>,
-        cwd: Option<&str>,
-        end: Option<&str>,
+        cwd: Option<&Path>,
+        env: Option<&str>,
     ) -> Result<Self, Error> {
         let mut err = null_mut() as *mut winpty_error_t;
 
-        let to_wstring = |s| WideCString::from_str(s).unwrap();
+        fn to_wstring(s: &(impl AsRef<OsStr> + ?Sized)) -> Vec<u16> {
```

Comment:
I think a type parameter would likely be much more readable than `impl Trait` here. Would also save us the parenthesis.